lib/error_tracker/integrations/phoenix.ex

```diff
@@ -111,4 +115,22 @@ defmodule ErrorTracker.Integrations.Phoenix do
   def handle_event([:phoenix, :live_view, _action, :exception], _, metadata, :no_config) do
     ErrorTracker.report({metadata.kind, metadata.reason}, metadata.stacktrace)
   end
+
+  def handle_event([:phoenix, :live_component, :update, :start], _, metadata, :no_config) do
+    ErrorTracker.set_context(%{
+      "live_view.component" => metadata.component
+    })
+  end
+
+  def handle_event([:phoenix, :live_component, :handle_event, :start], _, metadata, :no_config) do
+    ErrorTracker.set_context(%{
+      "live_view.component" => metadata.component,
+      "live_view.event" => metadata.event,
+      "live_view.event_params" => metadata.params
+    })
+  end
```

Comment:
I think this approach worked for Live views but is not going to work well for live components because live components live in the live view process, and the context is stored per-process.

Correct me if I'm wrong, but I think that if we have a live view with a live component on it and this events occur:

1. LiveView is loaded and mount is executed
2. Handle params is executed
3. The live view is rendered
4. The live component is updated or receives an event
5. an exception happened in the live view

The context will be something like:

```elixir
%{
  "live_view.view" => "MyApp.MyLiveView",
  "live_view.uri" => "URI of the liveview",
  "live_view.params" => %{ ... },
  "live_view.component" => "MyApp.MyLiveComponent",
  "live_view.event" => ":liveview_component_event",
  ...
}
```

Meaning that we will always have context of the last live component even if it didn't generate the exception.

In order to do this properly, we may need to think about this module and assign the context of LiveComponents when the exception occurs, as many live components can live in the same process. For live views is not the same case because we have only one per process.

Something like this may work

```suggestion
  def handle_event([:phoenix, :live_component, :update, :exception], _, metadata, :no_config) do
    ErrorTracker.report(
      {metadata.kind, metadata.reason}, 
      metadata.stacktrace,
      %{"live_view.component" => metadata.component}
    )
  end

  def handle_event([:phoenix, :live_component, :handle_event, :exception], _, metadata, :no_config) do
    ErrorTracker.report(
      {metadata.kind, metadata.reason}, 
      metadata.stacktrace,
      %{
        "live_view.component" => metadata.component,
        "live_view.event" => metadata.event,
        "live_view.event_params" => metadata.params
      }
    )
  end
```

(I wrote this directly on GitHub, I cannot assure it is correct, please check it :D )

With this kind of approach we can remove all live_component `:start` events and remove the exception case below this lines as it is no longer needed.

What do you think?

Reply:
Ah, yup. That makes a lotta sense. I can make that change.

Reply:
Alright, I _think_ the latest commit resolves this like you suggested. Let me know if any more changes are needed.

Though it does make me wonder; will there be similar issues in a LiveView? For example, if the following happened:

1. mount the liveview
2. handle_event("something", params, socket) in the liveview
3. handle_info("something_else", socket) causing a re-render
4. render -> exception occurs

I think `live_view.event` and `live_view.event_params` would still be in the error context when the exception was reported. They would have been assigned to context in step 2. But by step 4, that data could be irrelevant. Am I understanding that right?

Reply:
It depends, I think there is no universal solution on this.

The idea behind adding some context on the `:start` events is that:
- if you report something, you don't need to add any extra context
- if something else fails later, the latest event is in the context

However, that doesn't work out when we talk about multiple live components per process - a strange mix of events may occur.

That makes me think if we should stop doing this behaviour for all live view/component related events, or just for the live component events.

Any thoughts @crbelaus ?

(as for the changes made, they look perfect! ✨ thanks a lot for your time and effort)

Reply:
Yes, let's stop adding context in LiveView's handle_event start and instead just add it during handle_event exception just like we do for LiveComponents.

This should mitigate the problem as the process context would only have the LiveView name, URI and initial params. Everything else would be recorded only for exceptions.

Reply:
I've opened #78 to address this discussion.